data_structures/src/superblock.rs

```diff
@@ -499,27 +506,46 @@ pub fn calculate_superblock_signing_committee(
     signing_committee_size: u32,
     current_superblock_index: u32,
     superblock_hash: Hash,
+    block_epoch: Epoch,
 ) -> HashSet<PublicKeyHash> {
     // If the number of identities is lower than committee_size all the members of the ARS sign the superblock
     if ars_identities.len() < usize::try_from(signing_committee_size).unwrap() {
         ars_identities.identities
     } else {
         // Hash of the current_index, to avoid potential committee collisions
         let index_hash = Hash::from(calculate_sha256(&current_superblock_index.to_be_bytes()));
-        // Start counting the members of the subset from the superblock_hash plus superblock index hash
-        let mut first = u32::from(*superblock_hash.as_ref().get(0).unwrap())
-            + u32::from(*index_hash.as_ref().get(0).unwrap());
-        // We need to choose a first member from all the potential ARS members
-        first %= ars_identities.len() as u32;
-
-        // Get the subset
-        let subset = magic_partition(
-            &ars_identities.ordered_identities.to_vec(),
-            first.try_into().unwrap(),
-            signing_committee_size.try_into().unwrap(),
-        );
-        let hs: HashSet<PublicKeyHash> = subset.iter().cloned().collect();
-        hs
+        let first_byte_sb_hash = *superblock_hash.as_ref().get(0).unwrap();
+        let first_byte_index_hash = *index_hash.as_ref().get(0).unwrap();
+
+        if after_second_hard_fork(block_epoch, get_environment()) {
+            // Start counting the members of the subset from 'first_byte_index_hash'x256 + 'first_byte_sb_hash'
+            let mut first =
+                u64::from(first_byte_index_hash) * 256u64 + u64::from(first_byte_sb_hash);
+            // We need to choose a first member from all the potential ARS members
+            first %= ars_identities.len() as u64;
+
+            // Get the subset
+            magic_partition_2(
+                &ars_identities.ordered_identities.to_vec(),
```

Comment:
This `to_vec()` was not needed

```suggestion
                &ars_identities.ordered_identities,
```